### DISCUSSION TITLE
How to use non-default compiler for dependency compilation in vcpkg manifest mode?

### DISCUSSION THREAD

**Discussion by stripe2933** [microsoft/vcpkg#40007] [Q&A]
My system is macOS, and I want to use a dependency that can be compiled with Clang (from homebrew), but not with AppleClang. vcpkg uses AppleClang for dependency compilation by default, therefore I want to change this behavior.

What I tried is: create `clang-toolchain.cmake` file:

```cmake
set(CMAKE_C_COMPILER "/opt/homebrew/opt/llvm/bin/clang-18")
set(CMAKE_CXX_COMPILER "/opt/homebrew/opt/llvm/bin/clang++")
```

And pass file path as `VCPKG_CHAINLOAD_TOOLCHAIN_FILE`, like:

```sh
cmake -S . -B build -G Ninja -DCMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=<path/to/clang-toolchain.cmake>
```

However, installation still yields errors and log message says I'm still using AppleClang 15.

**Comment by autoantwort** [comment]
You have to create and use a custom triplet and set https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_chainload_toolchain_file inside it
